apps/meteor/app/federation-v2/server/Federation.ts

```diff
@@ -1,19 +1,40 @@
-import type { IRoom, ValueOf } from '@rocket.chat/core-typings';
-import { isDirectMessageRoom } from '@rocket.chat/core-typings';
+import type { IRoom, IUser, ValueOf } from '@rocket.chat/core-typings';
+import { isRoomFederated, isDirectMessageRoom } from '@rocket.chat/core-typings';
+import { Subscriptions } from '@rocket.chat/models';
 
 import { RoomMemberActions } from '../../../definition/IRoomTypeConfig';
 import { escapeExternalFederationEventId, unescapeExternalFederationEventId } from './infrastructure/rocket-chat/adapters/MessageConverter';
 
 const allowedActionsInFederatedRooms: ValueOf<typeof RoomMemberActions>[] = [
 	RoomMemberActions.REMOVE_USER,
+	RoomMemberActions.SET_AS_OWNER,
+	RoomMemberActions.SET_AS_MODERATOR,
 	RoomMemberActions.INVITE,
 	RoomMemberActions.JOIN,
 	RoomMemberActions.LEAVE,
 ];
 
 export class Federation {
-	public static actionAllowed(room: IRoom, action: ValueOf<typeof RoomMemberActions>): boolean {
-		return isDirectMessageRoom(room) && action === RoomMemberActions.REMOVE_USER ? false : allowedActionsInFederatedRooms.includes(action);
+	public static actionAllowed(room: IRoom, action: ValueOf<typeof RoomMemberActions>, userId?: IUser['_id']): boolean {
+		if (!isRoomFederated(room)) {
+			return false;
+		}
+		if (isDirectMessageRoom(room)) {
+			return false;
+		}
+		if (!userId) {
+			return true;
+		}
+
+		const userSubscription = Promise.await(Subscriptions.findOneByRoomIdAndUserId(room._id, userId));
+		if (!userSubscription) {
+			return true;
+		}
```

Comment:
That's more of a concern because there are a lot of unknowns calls for this method, and since I'm adding this new feature, I thought it would be better to keep as-was and then add the validation if and only if we have all the data we need for federation.